packages/doenetml-worker-rust/lib-doenetml-core/src/core/props/data_query/data_query.rs

```diff
@@ -89,43 +101,67 @@ impl From<LocalPropIdx> for PropSpecifier {
+    /// - `container`: the component whose children will be searched. For example, `PropComponent::Me`
+    /// to search your own children, or `PropComponent::Parent` to search your parent's children.
+    /// - `filter`: A composition of `ContentFilter`s. These can be combined with `Op::And`, `Op::Or`,
+    /// and `OpNot`.
     ///
-    /// For example, if the filter has one element `(PropProfile:Hidden`, `PropValue::Boolean(false))`, then
-    /// - if `include_if_missing_profile` is `true`,
-    ///   all component children that have a prop with the profile `Hidden` whose value is not `false`
-    ///   will be filtered out.
-    /// - if `include_if_missing_profile` is `false`,
-    ///   all component children that do not have a prop with the profile `Hidden` with value `false`
-    ///   will be filtered out.
-    FilteredChildren {
-        filters: Vec<DataQueryFilter>,
-        include_if_missing_profile: bool,
+    /// ## Example
+    /// ```rust
+    /// # use std::rc::Rc;
+    /// # use doenetml_core::props::{DataQuery, PropSource, ContentFilter, Op, PropProfile};
+    /// DataQuery::ComponentRefs {
+    ///    container: PropSource::Me,
+    ///    filter: Rc::new(Op::And(
+    ///      ContentFilter::IsType("section"),
+    ///      ContentFilter::HasPropMatchingProfile(PropProfile::Hidden),
+    ///    ))
+    /// }
+    /// # ;
+    /// ```
+    /// This query will return all children of the querying component that are `<section>` components
+    /// and have a prop matching the `PropProfile::Hidden` profile.
+    ComponentRefs {
+        /// Children of this component will be searched
+        container: PropSource,
+        /// How to filter the children. This should be a [`ContentFilter`] or a
+        /// composition of [`ContentFilter`]s.
+        ///
+        /// See [`DataQuery::ComponentRefs`] for an example.
+        filter: Rc<dyn for<'a> BindableAsGraphNodeFilter<'a>>,
+    },
+
+    AncestorRefs {
+        /// The component where the search starts.
+        /// This node will be included in the results if it matches the filter.
+        source: PropSource,
+        /// The filter to apply to the ancestors. Ancestors matching this filter
+        /// are returned.
+        ///
+        /// See [`DataQuery::ComponentRefs`] for an example of a filter.
+        filter: Rc<dyn for<'a> BindableAsGraphNodeFilter<'a>>,
     },
 
     /// Query for a particular prop of a component
     Prop {
-        /// If None, prop is from the component making the query.
-        component: PropComponent,
+        /// Where to look for the desired prop.
+        source: PropSource,
+        /// What prop to look for. Node `PropSpecifier::LocalIdx` is only valid
+        /// when used in conjunction with `PropComponent::Me`.
+        prop_specifier: PropSpecifier,
+    },
 
-        /// The prop from component_idx or component making the query.
+    /// Query multiple components and pick a prop from each.
+    PickProp {
```

Comment:
The fact that `Pick` implies multiple components is a bit confusing terminology. (It seems like that is the difference.) Any choice here is influenced by the larger discussion, above, on comparison to the cases without props.

Looking at this more, I guess the difference isn't just single component versus multiple components. In the case of `PickPropSource::NearestMatchingAncestor`, you still just get a single component, but the component selected is determined by the `PropSpecifier` (as opposed to the `Prop` query, where the component is first selected). The way this works seems quite similar to the filtering of components.

Reply:
This replaced the previous `ComponentProps`.

Ruby on Rails uses the function `pluck` for when you take one entry from each item in an array. SQL uses `select`. Preferences? `SelectProp` does sound good to anyone with a SQL background.